blocks/article-feed/article-feed.js

```diff
@@ -289,6 +290,20 @@ async function filterArticles(config, feed, limit, offset) {
     const beforeLoading = new Date();
     // eslint-disable-next-line no-await-in-loop
     const index = await fetchBlogArticleIndex();
+    if (config.pinned) {
+      const params = new URLSearchParams(window.location.search);
+      const articleFeedOverride = params.get('article-feed');
+      for (let i = config.pinned.length - 1; i >= 0; i -= 1) {
+        const pin = config.pinned[i];
+        const path = new URL(pin).pathname;
+        // eslint-disable-next-line no-await-in-loop
+        const article = articleFeedOverride ? await getBlogArticle(path) : index.byPath[path];
+        if (article) {
+          index.data.unshift(article);
```

Comment:
We probably need to remove the article from its original position to keep `data` clean (even though I think we check for duplicates before showing an article).